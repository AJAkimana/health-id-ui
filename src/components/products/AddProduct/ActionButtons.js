import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Button } from '@material-ui/core';
import { ActionButtonStyles } from '../../../assets/styles/products/addProductStyles';


const ActionButtons = (props) => {
  const { handleSendForApproval, handleAddAnotherProduct, disabled } = props;

  return (
    <Fragment>
      <Button
        disabled={disabled}
        variant="outlined"
        style={ActionButtonStyles.addButton}
        className="new-btn"
        onClick={handleAddAnotherProduct}
      >
      ADD NEW
      </Button>
      <Button
        disabled={disabled}
        variant="contained"
        color="primary"
        style={ActionButtonStyles.doneButton}
        className="create-btn"
        onClick={handleSendForApproval}
      >
    SEND FOR APPROVAL
      </Button>
    </Fragment>
  );
};

ActionButtons.propTypes = {
  handleAddAnotherProduct: PropTypes.func.isRequired,
  handleSendForApproval: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};

ActionButtons.defaultProps = {
  disabled: false
};

export default ActionButtons;
