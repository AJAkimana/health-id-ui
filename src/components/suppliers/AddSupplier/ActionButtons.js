import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Button } from '@material-ui/core';
import { ActionButtonStyles } from '../../../assets/styles/suppliers/addSupplierStyles';


const ActionButtons = (props) => {
  const { handleSendForApproval, handleAddAnotherSupplier, disabled } = props;

  return (
    <Fragment>
      <Button
        id="saveNew"
        disabled={disabled}
        variant="outlined"
        style={ActionButtonStyles.doneButton}
        className="new-btn"
        onClick={handleAddAnotherSupplier}
      >
      SAVE AND ADD NEW
      </Button>
      <Button
        id="save"
        disabled={disabled}
        variant="contained"
        color="primary"
        style={ActionButtonStyles.addButton}
        className="create-btn"
        onClick={handleSendForApproval}
      >
    SAVE
      </Button>
    </Fragment>
  );
};

ActionButtons.propTypes = {
  handleAddAnotherSupplier: PropTypes.func.isRequired,
  handleSendForApproval: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};

ActionButtons.defaultProps = {
  disabled: false
};

export default ActionButtons;
