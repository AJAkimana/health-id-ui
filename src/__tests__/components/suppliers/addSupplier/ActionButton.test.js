import React from 'react';
import { shallow } from 'enzyme';
import ActionButtons from '../../../../components/suppliers/AddSupplier/ActionButtons';

describe('Supplier Action Button', () => {
  const props = {
    handleAddAnotherSupplier: jest.fn(),
    handleSendForApproval: jest.fn(),
    disabled: true
  };

  it('should be defined', () => {
    expect(ActionButtons).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(<ActionButtons {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders disabled action buttons', () => {
    const props = {
      handleSendForApproval: jest.fn(),
      handleAddAnotherSupplier: jest.fn(),
      disabled: true
    };
    const wrapper = shallow(<ActionButtons {...props} />);
    expect(wrapper.find('.new-btn').length).toBe(1);
    expect(wrapper.find('.create-btn').length).toBe(1);
  });

  it('renders active action buttons ', () => {
    const props = {
      handleSendForApproval: jest.fn(),
      handleAddAnotherSupplier: jest.fn(),
      disabled: false
    };
    const wrapper = shallow(<ActionButtons {...props} />);
    expect(wrapper.find('.new-btn').length).toBe(1);
    expect(wrapper.find('.create-btn').length).toBe(1);
  });
});
