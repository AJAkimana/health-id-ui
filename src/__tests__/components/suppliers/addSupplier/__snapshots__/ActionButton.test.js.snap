// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Supplier Action Button should render correctly 1`] = `ShallowWrapper {}`;
